target/linux/ipq40xx/image/generic.mk

```diff
@@ -977,6 +977,20 @@ define Device/unielec_u4019-32m
 endef
 TARGET_DEVICES += unielec_u4019-32m
 
+define Device/zte_mf286d
+	$(call Device/FitzImage)
+	DEVICE_VENDOR := ZTE
+	DEVICE_MODEL := MF286D
+	SOC := qcom-ipq4019
+	DEVICE_DTS_CONFIG := config@ap.dk04.1-c1
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	KERNEL_SIZE := 4096k
```

Comment:
Is this a hard limit? Sure lzma'd FitzImage should be enough(tm), but it would be great to have this documented.

Reply:
I'm not sure. Stock u-boot looks like that:

`U-Boot 2012.07 [Chaos Calmer 15.05.1,f53cef1+r49254] (Mar 23 2020 - 11:33:39)

smem ram ptable found: ver: 1 len: 3
DRAM:  256 MiB
machid : 0x8010001
NAND:  ID = 302012c
Vendor = 2c
Device = 1
ONFI device found
SF: Detected GD25Q16 with page size 4 KiB, total 2 MiB
ipq_spi: page_size: 0x100, sector_size: 0x1000, size: 0x200000
130 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
machid: 8010001
flash_type: 0
Uaztemain: enter into !
ZTE_InitFotaFlashPara: nand_curr_device=0
zte_getHandOffState: read data=0x20 from 0x0
Hit any key to stop autoboot:  0
Creating 1 MTD partitions on "nand0":
0x000001800000-0x000003500000 : "mtd=0"
UBI: attaching mtd2 to ubi0
UBI: physical eraseblock size:   131072 bytes (128 KiB)
UBI: logical eraseblock size:    126976 bytes
UBI: smallest flash I/O unit:    2048
UBI: VID header offset:          2048 (aligned 2048)
UBI: data offset:                4096
UBI: attached mtd2 to ubi0
UBI: MTD device name:            "mtd=0"
UBI: MTD device size:            29 MiB
UBI: number of good PEBs:        232
UBI: number of bad PEBs:         0
UBI: max. allowed volumes:       128
UBI: wear-leveling threshold:    4096
UBI: number of internal volumes: 1
UBI: number of user volumes:     2
UBI: available PEBs:             70
UBI: total number of reserved PEBs: 162
UBI: number of PEBs reserved for bad PEB handling: 2
UBI: max/mean erase counter: 1/0
Read 0 bytes from volume kernel to 84000000
No size specified -> Using max size (4317184)
## Booting kernel from FIT Image at 84000000 ...
   Using 'config@ap.dk04.1-c1' configuration
   Trying 'kernel@1' kernel subimage
     Description:  ARM OpenWrt Linux-3.14.77
     Type:         Kernel Image
     Compression:  gzip compressed
     Data Start:   0x840000e4
     Data Size:    3616201 Bytes = 3.4 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: 0x80208000
     Entry Point:  0x80208000
     Hash algo:    crc32
     Hash value:   409a9b81
     Hash algo:    sha1
     Hash value:   1aad3f71030a9387e45ae1ebb85d29222e303878
   Verifying Hash Integrity ... crc32+ sha1+ OK
## Flattened Device Tree from FIT Image at 84000000
   Using 'config@ap.dk04.1-c1' configuration
   Trying 'fdt@ap.dk04.1-c1' FDT blob subimage
     Description:  ARM OpenWrt qcom-ipq40xx-ap.dkxx device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x843e7474
     Data Size:    38930 Bytes = 38 KiB
     Architecture: ARM
     Hash algo:    crc32
     Hash value:   6a16dabd
     Hash algo:    sha1
     Hash value:   d0b8da2df3282e3b8115fd4fbb83851ba56d4f81
   Verifying Hash Integrity ... crc32+ sha1+ OK
   Booting using the fdt blob at 0x843e7474
   Uncompressing Kernel Image ... OK
   Loading Device Tree to 86ff3000, end 86fff811 ... OK
eth0 MAC Address from ART is not valid
eth1 MAC Address from ART is not valid
Using machid 0x8010001 from environment

Starting kernel ...
`

I will try worse compression and >4MB kernel.

Reply:
10.6MB kernel boot successfully. I'll drop this 4M limit.